zookeeper-server/zookeeper-server-3.5.6/src/main/java/com/yahoo/vespa/zookeeper/Reconfigurer.java

```diff
@@ -88,10 +83,48 @@ private String connectionSpec(ZookeeperServerConfig config) {
         return String.join(",", servers(config));
     }
 
-    private List<String> servers(ZookeeperServerConfig config) {
+    private static List<String> servers(ZookeeperServerConfig config) {
         return config.server().stream()
                 .map(server -> server.hostname() + ":" + server.quorumPort() + ":" + server.electionPort())
                 .collect(Collectors.toList());
     }
 
+    ZookeeperServerConfig existingConfig() {
+        return zooKeeperRunner.zookeeperServerConfig();
```

Comment:
Is this ever updated Looks like it's only set when starting the server, so `shouldReconfigure` is always comparing the new config to the first config it received, not the current.

Reply:
You are right, that's wrong.